packages/client/base/src/services/api/BaseCrossmintService.ts

```diff
@@ -1,14 +1,16 @@
-import { CrossmintServiceError } from "@crossmint/client-sdk-base";
+import { CrossmintServiceError } from "@/error";
+import { SDKLogger } from "@/utils/SDKLogger";
+
 import { validateAPIKey } from "@crossmint/common-sdk-base";
 
-import { CROSSMINT_DEV_URL, CROSSMINT_PROD_URL, CROSSMINT_STG_URL } from "../utils/constants";
-import { logPerformance } from "../utils/log";
+import { CROSSMINT_DEV_URL, CROSSMINT_PROD_URL, CROSSMINT_STG_URL } from "../../consts";
 import { APIErrorService } from "./APIErrorService";
 
 export abstract class BaseCrossmintService {
     public crossmintAPIHeaders: Record<string, string>;
     protected crossmintBaseUrl: string;
-    protected apiErrorService: APIErrorService;
+    protected abstract apiErrorService: APIErrorService<string>;
+    protected abstract logger: SDKLogger;
```

Comment:
You consider injecting these two instead? I don't think there's a downside, and then you can use this w/o defining a subclass, which would be useful for SDKs that only hit a single endpoint or two, or that don't really need any special error handling.

Reply:
I played around with this a bit and I think abstract properties is a bit cleaner here. Dependency injection gets messy fast when not using a DI framework. Also keep in mind that BaseCrossmintService is abstract!